### DISCUSSION TITLE
Running on a non-Pi Linux Desktop

### DISCUSSION THREAD

**Discussion by Tweeks-va** [WheezyE/Winelink#48] [Q&A] [answered]
Post install, I'm a little iffy on how to bring it up correctly.

I'm running MXLinux (Anti-X Debian based), and your install script was great and seemed to set things up. The UIs popped to life (where doing it manually I just got winemono errors)... and the set up seemed to complete. However, where's the resulting executable stored now?  I have an icon in my drop down XFCE desktop launcher.. but it never comes up.  It there a way I should invoke it from the command line?  I assume it's not trying to run it on box86, since I'm already on an x86 laptop.. right?

T.Weeks

**Comment by WheezyE** [comment, marked as answer]
Hey there!  I'm kind of amazed that the RPiOS script worked on your x86 laptop. You are correct that box86 shouldn't be needed for an x86 laptop.  The wine installation might work though since it is made for i386 Debian.  The Windows executable files should be located within `~/.wine/drive_c/`. You will need to invoke wine to run them (for example: `wine ~/.wine/drive_c/VARA/VARA.exe`).

I've been working on adding other distros to the script in a patch branch on the github that you might try testing out.  I just added x64 Mint (Ubuntu) and x64 Debian https://github.com/WheezyE/Winelink/commit/97e34626cb22ee4c61b77add31382eb51c6a8272 though I haven't tested x86 yet (I need to install those OS's on my OracleVM first).
```
cd ~/Downloads
wget https://raw.githubusercontent.com/WheezyE/Winelink/97e34626cb22ee4c61b77add31382eb51c6a8272/install_winelink.sh
bash install_winelink.sh
```

**Reply by Root944** [reply]
WheezyE, I too have had recent success using your package installation script on a Linux x86_64 tablet, specifically a MS Surface GO (version 0), wiped of Windows and running Ubuntu 22.04.2, Kernel 5.19.3, on CPU Intel Pentium 4415Y.  Winlink Express, VARA, VARA FM, and VarAC each work reliably.  I am not a Linux expert and followed the guidance as if installing on a RPi.  The only bug I have is Winlink forms don't seem to accept input information correctly (inputs wind up beside the intended field vice in the intended field).  Separately, unrelated to your script, JS8Call works fine under CrossOver.

Thank you for putting this together even if a Linux x86 is not the intended use case.  The setup works nicely here.

One note/question:  I suspect the entire package is not friendly to modification.  I tried and failed to successfully install the latest update to VarAC.  Following the failure, I used Timeshift to reset my tablet to the state previous to the update attempt and all's well (except the Winlink form behavior mentioned above).  So I will leave well enough alone...

Thanks for the good work.  73, Mark

**Comment by Tweeks-va** [comment]
Cool.. I can get VARA to run now.. but with no radio hooked up.. I still seem to get a constant VU meter reading:

 
Why would this be?
 
I'm pretty good with low level Linux stuff.. but I'm juuuust starting off with this digital HD stuff (I'm still working on my General).. and was just trying to install Winlink on Linux so that I could create a Winlink account.  I'm not even sure how Winlink, VARA and these pther apps work together.  
 
I don't mind being a Guinnea pig for VARA, but need some kind of overview tutorial as to how it all goes together.  I get the concept of sound cards how digital dat gets encoded into PSK signals.. but I've just never done it before.  Let me know if you recommend any good guides.
 
T.Weeks
KK4GCWb
 
 
On Sunday, September 11, 2022 7:47pm, "Eric Wiessner (KI7POL)" ***@***.***> said:

Hey there! I'm kind of amazed that the RPiOS script worked on your x86 laptop. You are correct that box86 shouldn't be needed for an x86 laptop. The wine installation might work though since it is made for i386 Debian. The Windows executable files should be located within ~/.wine/drive_c/. You will need to invoke wine to run them (for example: wine ~/.wine/drive_c/VARA/VARA.exe).
I've been working on adding other distros to the script in a patch branch on the github that you might try testing out. I just added x64 Mint (Ubuntu) and x64 Debian 97e3462 though I haven't tested x86 yet (I need to install those OS's on my OracleVM first).
cd ~/Downloads
wget https://raw.githubusercontent.com/WheezyE/Winelink/97e34626cb22ee4c61b77add31382eb51c6a8272/install_winelink.sh
bash install_winelink.sh

**Comment by WheezyE** [comment]
No worries 🙂 I’m a tech too. I’ll probably work on getting my general some
day just been busy. I found YouTube to be very helpful with learning the
overview of all the menus in RMS Express. You can practice sending Winlink
messages with RMS Express using the “telnet” option (which feels very
similar to using RMS Express like an email client - feels like gmail or
outlook).  VARA and ARDOP are separate programs, but talk to RMS Express
and then send audio to your radio. I just put headphones on a USB soundcard
on my Pi and listen to the tones - ive never actually sent a message myself
with VARA.

I’m not sure about the VU meter but my guess would be your audio in has
some static or lots of noise in it or something like that.

I’ll put MXLinux on my roadmap for distros to support later.

73
Eric

**Comment by Tweeks-va** [comment]
What's the diff between WinLink and RMSExpress?
 
If you support Debian.. it will probably work on MXLinux. It's basically Debian without systemd.
 
I thought your script installed WinLink.. but all I see is RMSExpress.
 
If you have any good overview videos.. please send me a good link. :)
 
Tweeks

**Comment by WheezyE** [comment]
> What's the diff between WinLink and RMSExpress?

As I understand it, Winlink is like saying "AT&T"/"Verizon". RMSExpress/Pat are like saying "Outlook"/"Gmail". VARA/ARDOP are like "5G"/"2G" (or maybe more like "dialup").
 - Winlink is the umbrella term for the entire communication network effort (people, computers, software, and radios).
 - RMS Express is a messaging client made for Windows. Pat also has the same functionality and runs natively on Linux. It controls radios (with CAT control) and tells modems to play/receive audio at the right times (and what content to send).
 - VARA and ARDOP are "modem" _(modulation-demodulation)_ protocols for encoding data into sound (hams call them "TNC's" instead of modems for some unknown reason). They take text input, convert it to audio, and then your radio plays the audio over the air.

> If you support Debian.. it will probably work on MXLinux. It's basically Debian without systemd.

It hopefully will, but ... the devil is in the details, as they say.  I'll try to test it, but feel free to try (or modify and PR) the current Debian script in the patch-5 branch if you beat me to it.

> If you have any good overview videos.. please send me a good link. :)

I don't know of any overview videos off-hand, but [this one](https://www.youtube.com/watch?v=nD5FfNIJaCo) looks pretty comprehensive?

**Comment by Schlad32** [comment]
Hello, This script works great on a few different model laptops running Linux Mint. However, I was testing Zorin-OS (another debian flavor), and I get the "No install path has been scouted for this distro yet" error.

Could this distro be added? Or is there a work around in the meantime? File structure seems to be the same as the rest of the debian flavors.

Thanks for your help and all your contributions!!
